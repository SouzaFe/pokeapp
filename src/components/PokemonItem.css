
.div-item-list {
    position: relative;
    list-style-type: none;
    border: 1px solid gray;
    border-radius: 4px;
    width: 161px;
    height: 100px;
    margin: 5px;
    display: inline-block;
    text-transform: capitalize;
    text-align: center;
    box-shadow: 3px 3px 6px 1px #ccc;
    cursor: pointer;
    background-color: #f3f3f3;
    transition: all .13s ease-in-out;
    font-size: 12px;
}

.div-item-list:hover {
    transform: scale(1.04);
}

.div-item-list > a {
    text-decoration: none;
    color: #000000;
    height: 80px;
    margin: auto;
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
}

.div-item-list > a > img {
    margin: -10px 0 -8px 0;
    width: 75px;
    height: 75px;
}

.div-item-list > a > p {
    font-weight: bold;
    margin: 0;
}