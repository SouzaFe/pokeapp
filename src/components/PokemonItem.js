import React from 'react';
import './PokemonItem.css';
import { Link } from "react-router-dom";

const PokemonItem = (props) => {
    let url = props.url.replace(/\/$/, "");
    url = url.substr(url.lastIndexOf("/") + 1);
    const urlImage = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${url}.png`;

    return (
        <li className="div-item-list">
            <Link to={{ pathname: `/Pokemon/${props.name}`, state: { toBack: props.toBack } }}>
                <img alt={props.name} src={urlImage} />
                <p>
                    {props.name}
                </p>
            </Link>
        </li>);
}

export default PokemonItem;