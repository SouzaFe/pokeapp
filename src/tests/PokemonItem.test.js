import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";

import PokemonItem from "../components/PokemonItem";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("Gerar o snapshot do componente", () => {
    act(() => {
        render(
            <Router>
                <Switch>
                    <PokemonItem name="pikachu" url="https://pokeapi.co/api/v2/pokemon/25/" toBack="https://pokeapi.co/api/v2/pokemon?limit=20&offset=20" />
                </Switch>
            </Router>, container);
    });
    expect(container.innerHTML).toMatchSnapshot();
});
