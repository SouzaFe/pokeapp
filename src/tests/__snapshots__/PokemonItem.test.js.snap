// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gerar o snapshot do componente 1`] = `"<li class=\\"div-item-list\\"><a href=\\"/Pokemon/pikachu\\"><img alt=\\"pikachu\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png\\"><p>pikachu</p></a></li>"`;
